abdiff/config.py

```diff
@@ -4,7 +4,12 @@
 
 
 class Config:
-    REQUIRED_ENV_VARS = ("WORKSPACE",)
+    REQUIRED_ENV_VARS = (
+        "AWS_ACCESS_KEY_ID",
+        "AWS_SECRET_ACCESS_KEY",
+        "AWS_SESSION_TOKEN",
```

Comment:
We could address this later perhaps, but wondering if there might be other ways to get AWS credentials to the container.

In the past, I've enjoyed/preferred using a mount where I mount my `~/.aws` directory to the container's `/root/.aws` directory.  This was I can use my SSO set credentials without ever getting these tokens, that expire.

Not proposing that way is _better_, but might be worth some thinking / research on ways to have a container have AWS credentials.

Reply:
Or.... what if we download the file to the host machine, and then mount that in the container for use?  This might have an advantage: for any given file the container would be reading from S3, it's going to happen _twice_ given the A/B nature of this application.

If we downloaded locally, we'd only pay the network penalty once for downloading the file to the host machine.  Then we'd mount for the container to use, it transforms from it, and returns a transformed file.  Once all containers are done (or more surgically, when all containers that require _that_ file are done) we could delete from the host machine.  Or heck, we could even have an option to keep it around if we know we're going to re-run some transformations.  

This adds a bit of complexity, but I think it would halve the time we spend on network reading.  And, as mentioned above, it might open the option for not deleting the files we've retrieved if we know we're doing more runs.  I'd happily allow some XML/JSONLines file to accumate on my host machine if I know it would dramatically increase the runtimes for the A/B diffs.  Happy to discuss more.

We could also make a ticket for this, and just keep as-is.  But FWIW, this would remove the need to pass AWS credentials to the container, which has advantages as well.

Reply:
On the flipside, perhaps it would make sense to revist later?  I see two things I'd like this application to be able to do eventually, which require some tweaks to this step:

1. read input files from "local" host machine, relative to the docker containers
2. run ab-diff on _two, distinct_ inputs sets of files_ for _one_ version of Transmog

Reply:
As discussed via Slack huddle, it was decided to **leave the method for passing AWS credentials as-is** -- creating a `.env` file -> copy-pasting credentials -> relying on `Config` to load environment variables as needed in the call to `run_ab_transforms`. While the A/B transform workflow will be run locally, minimizing security risks, it's generally recommended to avoid mounting files with sensitive info into Docker containers.

As for the latter comments on updating method for accessing files, it was decided to proceed with the current approach of transforming input files from S3 as a "first pass" version of this command. Ticket [TIMX-353](https://mitlibraries.atlassian.net/browse/TIMX-353) has been created to explore this topic further.

Reply:
Sounds good!  Thanks for considering and the summary.